common/types/overflow.go

```diff
@@ -326,6 +326,26 @@ func int64ToUint64Checked(v int64) (uint64, error) {
 	return uint64(v), nil
 }
 
+// int64ToInt8Checked converts an int64 to an int8 value.
+//
+// If the conversion fails due to overflow the error return value will be non-nil.
+func int64ToInt8Checked(v int64) (int8, error) {
```

Comment:
Within the `int_test.go` there are some additional tests for overflows based on the Go-native integer width, e.g. `TestIntConvertToNative_Int32`. Would you mind adding similar tests 8-bit and 16-bit int conversions? Similarly for `uint_test.go`.

Thanks!

Reply:
Done. Thanks for taking the time to review these changes!

Reply:
Thanks for taking the time to make them!